mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala

```diff
@@ -94,8 +110,7 @@ private[ml] class WeightedLeastSquares(
       if (standardizeFeatures) {
         lambda *= aVar(j - 2)
       }
-      if (standardizeLabel) {
-        // TODO: handle the case when bStd = 0
+      if (standardizeLabel && bStd != 0) {
```

Comment:
The `WeightedLeastSquares` class is private and its instantiated in `LinearRegression` class where `standerizeLabe` parameter is hard wired to be `true`. So, the user doesn't have any control on this parameter.

We can throw an exception when `yStd` is zero and `regParam` is non-zero. But, if that is the case, then, why not to throw exception when `yStd` is zero regardless of what other parameters are? I cannot think of any interpretation of the model in this case.

The option could be to simply log a warning when we don't standardize the label here.

Let me know what you think.